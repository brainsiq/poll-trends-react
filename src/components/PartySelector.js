import React, { Component } from 'react';

class PartySelector extends Component {
  constructor() {
    super();

    this.state = { companies: [] };
  }

  loadCompanies() {
    return fetch('http://localhost:3001/companies')
      .then(response => response.json())
      .then(response => {
        const companies = response.map(company => Object.assign({
          selected: company.id === 'YG'
        }, company))
        this.setState({ companies });
      })
      .catch(err => {
        console.error(err);
      });
  }

  componentDidMount() {
    this.loadCompanies();
  }

  render() {
    return (
      <div>
        <label htmlFor="pollCompanies">Pollster</label>

        <select name="pollCompanies">
          {
            this.state.companies.map(company => <option value={company.id} key={company.id} selected={company.selected}>{company.name}</option>)
          }
        </select>
      </div>
    )
  }
}

export default PartySelector;
