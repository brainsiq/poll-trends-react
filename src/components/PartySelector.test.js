import React from 'react';
import renderer from 'react-test-renderer';
import PartySelector from './PartySelector.js';

beforeAll(() => {
  const companiesResponse = [{
    id: 'YG',
    name: 'YouGov'
  }, {
    id: 'SRVN',
    name: 'Survation'
  }];

  fetch.mockResponse(JSON.stringify(companiesResponse));
});

it('renders the select list with companies from the API', () => {
  const component = renderer
    .create(<PartySelector />);

  return component.getInstance().loadCompanies()
    .then(() => {
      expect(component.toJSON()).toMatchSnapshot();
    });
});
