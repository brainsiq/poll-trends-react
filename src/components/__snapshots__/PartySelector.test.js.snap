// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the select list of companies 1`] = `
<div>
  <label
    htmlFor="pollCompanies"
  >
    Pollster
  </label>
  <select
    name="pollCompanies"
  >
    <option
      selected={true}
      value="YG"
    >
      YouGov
    </option>
    <option
      selected={false}
      value="SRVN"
    >
      Survation
    </option>
  </select>
</div>
`;
